From: Carlos Garnacho <carlosg@gnome.org>
Date: Wed, 12 Jun 2024 23:04:53 +0200
Subject: build: Depend on parser modules for the internal common library

Sometimes in CI, the docs builder gets ahead of itself, before
the parser modules are even built. Set up this dependency between
the common library and the parser modules, in order to ensure these
modules are built before there's ever a chance they get used.
---
 src/libtracker-common/meson.build | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/libtracker-common/meson.build b/src/libtracker-common/meson.build
index 9355d1c..3cb0054 100644
--- a/src/libtracker-common/meson.build
+++ b/src/libtracker-common/meson.build
@@ -8,7 +8,7 @@ tracker_common_sources = [
 ]
 
 if unicode_library_name == 'icu'
-  libtracker_parser_libicu = shared_module('tracker-parser-libicu',
+  libtracker_parser = shared_module('tracker-parser-libicu',
     'tracker-parser-libicu.c',
     'tracker-parser-utils.c',
     'tracker-language.c',
@@ -23,7 +23,7 @@ if unicode_library_name == 'icu'
     name_suffix: 'so',
   )
 else
-  libtracker_parser_libunistring = shared_module('tracker-parser-libunistring',
+  libtracker_parser = shared_module('tracker-parser-libunistring',
     'tracker-parser-libunistring.c',
     'tracker-parser-utils.c',
     'tracker-language.c',
@@ -48,6 +48,7 @@ endif
 
 libtracker_common = static_library('tracker-common',
   tracker_common_sources,
+  libtracker_parser,
   dependencies: [tracker_common_dependencies, gmodule],
   c_args: [
     '-DPRIVATE_LIBDIR="@0@"'.format(tracker_internal_libs_dir),
